instance/models/server.py

```diff
@@ -449,3 +457,17 @@ def terminate(self):
                 self._status_to_unknown()
         else:
             self._status_to_terminated()
+
+    def _shutdown(self, os_server, poll_interval=10, max_wait=600):
+        """
+        Shutdown the server and wait to return until shutdown or wait threshold reached.
+
+        We don't have an explicit state for this and don't catch exceptions, which is why this is private.
+        The caller is expected to handle any exceptions and retry if necessary.
+        """
+        os_server.stop()
+        while max_wait and os_server.status != 'SHUTOFF':
```

Comment:
@UmanShahzad Just noticed this, why don't we post a thread for [terminate()](https://github.com/open-craft/opencraft/pull/359/files#diff-c40bbeb7cb644b7f9da62d92619b34e6R428) instead of keeping the process waiting for it to be completed after a certain amount of time?

Reply:
Commented on the ticket.